### ISSUE TITLE
Relocate to the Gitlab Xonotic org

### ISSUE THREAD

**Issue by garymoon** [MarioSMB/modpack#47]
Hi all!

I would like to propose the modpack be moved to the xonotic org on Gitlab. It would facilitate greater integration with the game itself, and make it more accessible to anyone contributing there also.

I would also like to work on the following changes:
* Make it an optional module in the game build scripts (`./all`)
  * This would also free us from the submodule duplication of xonotic-data
* Add a Gitlab CI pipeline
  * Something similar to the [Github Action](https://github.com/garymoon/modpack/blob/add_gh_action/.github/workflows/build.yml)
  * Perhaps ideally integrated with or triggered by existing pipelines
  * The Travis CI pipeline will shortly become defunct regardless as they cripple OSS builds
* Add some kind of artifact/release storage for users to download pre-compiled copies from
  * Perhaps integration with beta.xonotic.org or similar is possible
* Improvements to the existing build workflow
  * Some thoughts already at #46
  * Possibly made a moot point by the first proposal above

@MarioSMB I would especially appreciate your input on this if you're up to it please, and thank you! <3

**Comment by LegendaryGuard** [comment]
I agree with what it says here, it would improve the ways of working and save some time and effort when we use the tools every time. It would be great to advance this progress.

**Comment by MarioSMB** [comment]
Thank you for your interest in this project. As it stands currently, I am on hiatus from Xonotic-related development, so any plans will likely be put on hold for some time.  
That said, I appreciate the effort put into this changelist, and would certainly love to see some of them happen. Whether that happens sooner or later remains to be seen, it is unfortunate that Travis may be forcing some hands soon...  

It's too early to say of course, but my current affirmation is that it would be worthwhile to move a majority of the features out of this mod and into the core game (including the likes of piggybacking and zombie apocalypse), where they can be maintained along with the rest of the game and benefit a wider audience. Anything remaining in the mod would then be considered deprecated (such as the additional non-GPL content and the vehicles from other unnamed forks).  
There is a lot to consider here, though this line of thought stems from my no longer being a server administrator, so the desire to maintain mods is of course waning as I can no longer make use of this project.

**Comment by garymoon** [comment]
Thanks for the reply @MarioSMB :blue_heart: 

Just to clarify, I did not intend to suggest you start this work, but rather that in light of your hiatus I'd like to have a shot at it and am looking for your blessing.

I also agree that mainlining much of the mod would be beneficial and would happily see that happen. Indeed much of this change list would be steps along a path to that end. As a current server admin I'd be very disappointed to see the modpack fade from the scene.

If you're liking the look of the change list, please let us know, along with any modifications you'd like us to consider as well as to what extent (if any) you'd like to be consulted along the way.

Thanks Mario!

**Comment by MarioSMB** [comment]
Pardon the delay, I've had a lot to consider over this time.

First off, I'd like to start by stating the original reasoning behind splitting the modpack into a separate repository:  
- The features to be included in the mod are intended to never be merged into the base game, some even make use of non-GPL assets, so they don't really belong in the main data repository of Xonotic
- The xonotic-data submodule is an unfortunate large inclusion, but Xonotic Git is much larger and somewhat less accessible. Being able to run the mod on non-Git servers has been extremely valuable in the past
- Ease of maintenance; in the early days, the mod was included as a branch on Xonotic's data repository. The burden of keeping it in sync with master (among the hundreds of merge conflicts and codebase changes) was nearing a breaking point and the mod would have died had it stayed there. As a separate repository and a secluded codebase, syncing with master is often a 5 minute job, even after many months of neglect

Those points are the reason TimePath created the modpack system we have currently. It wasn't intended to last forever, but it has helped me develop the mod without being overwhelmed.  

Moving forward, I do believe a lot of what is proposed here is a good path to integration.  
I'd also like to suggest an alternative path to consider: bring the mod back as a branch in xonotic-data, but keep the code separate (in its own mod directory). The branch would have the mod's code enabled automatically, so contributors and admins would simply need to check out the branch to make use of it.  

I look forward to hearing your feedback here, and would be happy to help with either approach wherever I can.

**Comment by garymoon** [comment]
@LegendaryGuard has started integrating some parts into xonotic-data for his own purposes at https://gitlab.com/xonotic/xonotic-data.pk3dir/-/tree/LegendaryGuard/cyber. A conversation can be started to work out which parts should be MR'd.
